src/object_accessor.hpp

```diff
@@ -98,37 +98,76 @@ void Object::set_property_value_impl(ContextType& ctx, const Property &property,
     switch (property.type & ~PropertyType::Nullable) {
         case PropertyType::Object: {
             ContextType child_ctx(ctx, property);
-            auto link = child_ctx.template unbox<RowExpr>(value, true, try_update);
-            table.set_link(col, row, link.get_index(), is_default);
+            // Check if we are linking to an object already
+            auto curr_link = table.get_link(col,row);
+            auto link = child_ctx.template unbox<RowExpr>(value, true, try_update, update_only_diff);
+            if (!update_only_diff || curr_link != link.get_index()) {
+                table.set_link(col, row, link.get_index(), is_default);
+                ctx.did_change();
+            }
             break;
         }
-        case PropertyType::Bool:
-            table.set(col, row, ctx.template unbox<bool>(value), is_default);
+        case PropertyType::Bool: {
+            auto new_val = ctx.template unbox<bool>(value);
```

Comment:
This is a lot of duplicated code for each case that looks pretty easy to extract to a function.

Reply:
I thought about it, but was not really sure if it would be an improvement, but as you mention it, I will extract into a function.